File: edu/university/src/main/java/com/rkarp/uni/modules/wise15/rechnerstrukturen/uebung12/aufg2/Fah2Cel.java
package com.rkarp.uni.modules.wise15.rechnerstrukturen.uebung12.aufg2;

/**
 * Converts Fahrenheit to Celsius.
 * Formula: C = (F - 32) * 142/256
 */
public class Fah2Cel {
    public static int fah2cel(int fah) {
        return ((fah - 32) * 142) >> 8;
    }
}

File: edu/university/src/main/java/com/rkarp/uni/modules/wise15/rechnerstrukturen/uebung12/aufg2/Fah2Cel.kt
package com.rkarp.uni.modules.wise15.rechnerstrukturen.uebung12.aufg2

/**
 * Converts Fahrenheit to Celsius.
 * Formula: C = (F - 32) * 142/256
 */
class Fah2Cel {
    fun fah2cel(fah: Int): Int {
        return (fah - 32) * 142 shr 8
    }
}
